serpentTools/parsers/results.py

```diff
@@ -1,15 +1,319 @@
+                        self.metadata[keyName] = values
+                    elif self._idxSect[1] and self._checkAddVariable(currKey.group()):
+                        self.resdata[keyName][self._counter['step']-1, :] = values
+                    else:
+                        self._processUnivData(flagStore, currKey.group(), values)
+
+
+    def _processUnivData(self, flagStore, varName, values):
+        """Creates a univ container and stores the data"""
+        if flagStore:
+            self._currUniv = values
+            if self._daysStr in self.resdata.keys() and self._buStr in self.resdata.keys():
+                idxStep = self._counter['step']-1
+                self._days = self.resdata[self._daysStr][idxStep, 0]
+                self._bu = self.resdata[self._buStr][idxStep, 0]
+                self._buIdx = self.resdata[self._buIdxStr][idxStep, 0]
+                self._brachID = (self._currUniv, self._bu, self._buIdx, self._days)
+            self.univdata[self._brachID] = HomogUniv(self._currUniv, self._bu, self._buIdx, self._days)
+        else:
+            if self._checkAddVariable(varName):
+                if self.strSearch['inf'] in varName[0:4] or self.strSearch['b1'] in varName[0:4]:
+                    vals, uncs = splitItems(values)
+                    self.univdata[self._brachID].addData(varName, array(uncs), True)
+                    self.univdata[self._brachID].addData(varName, array(vals), False)
+                else:
+                    self.univdata[self._brachID].addData(varName, array(values), False)
+
+    def _preProcessData(self):
+        """Pre-process data before collecting it"""
+        self._versionSerpent()  # Get the utilized Serpent version
+        self._numUnivSteps()  # Count the number of universes repetitions for memory allocation
+        vals = [self._univID[item] for item in self._univID]
+        if vals == {}:
+            raise SerpentToolsException("No universes are present in the file.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        if len(vals) > 1 and not vals[1:] == vals[:-1]:
+            raise SerpentToolsException("The repetitions of universes in the file is not equal.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        self._ChunkPosition()  # identify the 1st block for defining the dictionary
+        if self._ChunkPos == 0:
+            raise SerpentToolsException("The results file is corrupted.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+        self._defineResultDict()  # Obtain the names of the variables to be stored
+        if self._resKeys == []:
+            raise SerpentToolsException("No variables appear in the results file.\n"
+                          "Parser: {}\nFile: {}".format(self, self.filePath))
+
+    def _defineResultDict(self):
+        """ Initializes the dictionary for ALL results included in .res.m"""
```

Comment:
I'm not convinced we need to pre-allocate values in the results dictionary for __all__ variables, even in the preprocess stage. We already have logic for checking if variables, `_checkAddVariable`, should be added, which you use in other places.